### PR TITLE
Adding backspace motion in visual mode

### LINKED ISSUE #3220: Backspace doesn't move cursor back in visual mode (like vim)
Backspace doesn't move the cursor backwards in 'visual mode' like vim does, it should perform the same action as left cursor key, useful for altering your visual selection.

Note it works as expected in 'normal' mode

Thanks

### PR DESCRIPTION
**Pull request by pedroapfilho**
Fixes #3220

### CODE REVIEW

**Comment by J-Fields** [comment]
Thanks! Could you add support for `VisualBlock` mode as well?
Also, a few basic test cases would be awesome.

**Comment by pedroapfilho** [comment]
Of course I can! Just to know, on VisualBlock, it should select the last block, right?

**Comment by J-Fields** [comment]
Not quite sure what you mean tbh. Backspace should work the same as the left arrow in visual block mode

File: src/actions/motion.ts [hunk 1]
```diff
@@ -434,6 +434,16 @@ class BackSpaceInNormalMode extends BaseMovement {
   }
 }
 
+@RegisterAction
+class BackSpaceInVisualMode extends BaseMovement {
+  modes = [ModeName.Visual, ModeName.VisualBlock];
+  keys = ['<BS>'];
+
+  public async execAction(position: Position, vimState: VimState): Promise<Position> {
+    return position.getLeftThroughLineBreaks();
```
Comment (J-Fields) [inline src/actions/motion.ts]: It should only go through line breaks if `whichwrap` contains `b`
http://vimdoc.sourceforge.net/htmldoc/options.html#'whichwrap'

Comment (pedroapfilho) [reply src/actions/motion.ts]: Done!

**Pull request merged**